backend/qemu.pm

```diff
@@ -486,7 +486,10 @@ sub find_ovmf () { first { -e } @bmwqemu::ovmf_locations }
 
 sub virtio_console_names () {
     return () unless $bmwqemu::vars{VIRTIO_CONSOLE};
-    return map { 'virtio_console' . ($_ || '') } (0 .. ($bmwqemu::vars{VIRTIO_CONSOLE_NUM} // 1));
+    return (
+        'virtio_console', 'virtio_console_user',
+        map { 'virtio_console' . $_ } (1 .. ($bmwqemu::vars{VIRTIO_CONSOLE_NUM} // 1) - 1),
```

Comment:
I would not use VIRTIO_CONSOLE_NUM here, it will be added on wicked tests where it's not needed.
Either have it always defined with VIRTIO_CONSOLE, I think it should work.
Or enable it with e.g.  VIRTIO_CONSOLE_USER ?

Reply:
This is used for qemu
starting: /usr/bin/qemu-system-x86_64 -device VGA,edid=on,xres=1024,yres=768 -only-migratable -chardev ringbuf,id=serial0,logfile=serial0,logappend=on -serial chardev:serial0 -audiodev none,id=snd0 -device intel-hda -device hda-output,audiodev=snd0 -global isa-fdc.fdtypeA=none -m 4096 -cpu host -netdev user,id=qanet0 -device virtio-net,netdev=qanet0,mac=52:54:00:12:34:56 -object rng-random,filename=/dev/urandom,id=rng0 -device virtio-rng-pci,rng=rng0 -boot order=c -device qemu-xhci -device usb-tablet -smp 4 -enable-kvm -no-shutdown -vnc :95,share=force-shared -device virtio-serial -chardev pipe,id=virtio_console,path=virtio_console,logfile=virtio_console.log,logappend=on -device virtconsole,chardev=virtio_console,name=org.openqa.console.virtio_console -chardev pipe,id=virtio_console_user,path=virtio_console_user,logfile=virtio_console_user.log,logappend=on -device virtconsole,chardev=virtio_console_user,name=org.openqa.console.virtio_console_user -chardev socket,path=qmp_socket,server=on,wait=off,id=qmp_socket,logfile=qmp_socket.log,logappend=on -qmp chardev:qmp_socket -S -device virtio-scsi-pci,id=scsi0 -blockdev driver=file,node-name=hd0-overlay0-file,filename=/var/lib/openqa/pool/5/raid/hd0-overlay0,cache.no-flush=on -blockdev driver=qcow2,node-name=hd0-overlay0,file=hd0-overlay0-file,cache.no-flush=on,discard=unmap -device virtio-blk,id=hd0-device,drive=hd0-overlay0,bootindex=0,serial=hd0 -blockdev driver=file,node-name=cd0-overlay0-file,filename=/var/lib/openqa/pool/5/raid/cd0-overlay0,cache.no-flush=on -blockdev driver=qcow2,node-name=cd0-overlay0,file=cd0-overlay0-file,cache.no-flush=on,discard=unmap -device scsi-cd,id=cd0-device,drive=cd0-overlay0,serial=cd0
```

Reply:
I'm also to include this here. 
But we need to change https://github.com/cfconrad/os-autoinst/blob/master/doc/backend_vars.asciidoc
that `1` console is not `1` console but `2`. And a number `>1` is actualy always `$VIRTIO_CONSOLE_NUM +1` because we have now a `root` and a user `sticky` console.

Reply:
Well, `VIRTIO_CONSOLE_NUM` is just for `root-virtio-terminal`. `VIRTIO_CONSOLE_NUM=2` adds new `root-virtio-terminal`, that is how is being used.

I was thinking about changing `VIRTIO_CONSOLE_NUM` to `VIRTIO_CONSOLE_ROOT_NUM` and add new variable `VIRTIO_CONSOLE_USER_NUM` => to be able to add more user consoles, bug then I keep things as they are, because only wicked is using `VIRTIO_CONSOLE_NUM` (for uploading logs) and only HPC is using user console now. Who would need more user consoles?

Also, we're not _adding_ but _fixing_ `user-virtio-terminal`, it was added in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14967

I'm trying to minimize changes in these 3 PRs, so that it can be reasonably quickly tested and merged. Otherwise I would think about changing `VIRTIO_CONSOLE_NUM` name and/or functionality.